lemminx-maven/src/main/java/org/eclipse/lemminx/maven/MavenProjectCache.java

```diff
@@ -107,12 +105,41 @@ public MavenProject getLastSuccessfulMavenProject(DOMDocument document) {
 	private void check(DOMDocument document) {
 		Integer last = lastCheckedVersion.get(URI.create(document.getTextDocument().getUri()));
 		if (last == null || last.intValue() < document.getTextDocument().getVersion()) {
-			parse(document);
+			parseAndCache(document);
 
 		}
 	}
 
-	private void parse(DOMDocument document) {
+	public Optional<MavenProject> getSnspahotProject(File file) {
```

Comment:
Small typo, should be `getSnapshotProject(File file)`

Reply:
Thanks

Reply:
Fixed in new commit